src/main/java/seedu/address/logic/commands/EditEventCommand.java

```diff
@@ -0,0 +1,133 @@
+    private final Index index;
+
+    /**
+     * @param editEventDescriptor details to edit the event with
+     */
+    public EditEventCommand(EventDescriptor editEventDescriptor) {
+        requireNonNull(editEventDescriptor);
+
+        this.index = editEventDescriptor.getIndex().get();
+        this.editEventDescriptor = new EventDescriptor(editEventDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        // TODO
+        requireNonNull(model);
+        List<Event> eventList = model.getEvents();
+
+        if (index.getZeroBased() >= eventList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Event eventToEdit = eventList.get(index.getZeroBased());
+        Event editedEvent = createEditedEvent(eventToEdit, editEventDescriptor);
+
+        if (!model.hasEvent(eventToEdit)) {
+            throw new CommandException(MESSAGE_INVALID_EVENT);
+        }
+
+        if (!DateTime.isValidInterval(editedEvent.getStartDateTime(), editedEvent.getEndDateTime())) {
+            throw new CommandException(MESSAGE_INVALID_INTERVAL);
+        }
+
+        model.setEvent(eventToEdit, editedEvent);
+        return new CommandResult(String.format(MESSAGE_EDIT_EVENT_SUCCESS, editedEvent));
+    }
+
+    public EventDescriptor getEditEventDescriptor() {
+        return this.editEventDescriptor;
+    }
+
+    /**
+     * Creates a new edited event to replace the original
+     *
+     * @param eventToEdit
+     * @param editEventDescriptor
+     * @return the new Event to be set
+     */
+    protected static Event createEditedEvent(Event eventToEdit, EventDescriptor editEventDescriptor) {
+        assert eventToEdit != null;
+
+        Description description = editEventDescriptor.getDescription().orElse(eventToEdit.getDescription());
+        DateTime startDateTime = editEventDescriptor.getStartDateTime().orElse(eventToEdit.getStartDateTime());
+        DateTime endDateTime = editEventDescriptor.getEndDateTime().orElse(eventToEdit.getEndDateTime());
+        Recurrence recurrence = editEventDescriptor.getRecurrence().orElse(eventToEdit.getRecurrence());
+
+        if (recurrence.isRecurring()) {
+            return new RecurringEvent(description, startDateTime, endDateTime, recurrence);
+        } else {
+            return new OneTimeEvent(description, startDateTime, endDateTime);
+        }
+    }
+
+    /**
+     * Returns true if the two events have the same identity and data fields.
+     * This defines a weaker notion of equality between two events.
+     */
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof EditEventCommand)) {
+            return false;
+        }
+
+        EditEventCommand otherCommand = (EditEventCommand) other;
+        return editEventDescriptor.equals(otherCommand.editEventDescriptor);
+    }
+}
```

Comment:
```suggestion
    public boolean equals(Object other) {
        return other == this
          || (other instanceof EditEventCommand
          && this.editEventDescriptor.equals((EditEventCommand) other.editEventDescriptor));
    }
}
```

Thi is just so that the equals syntax is the same as all the other equals method in the project.